include/gpio.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file gpio.hpp
+ *
+ * @brief HAL GPIO class header
+ *
+ * @date 09/2024
+ */
```

Comment:
```suggestion
/**
 * @file gpio.hpp
 *
 * @brief HAL GPIO class header
 */
```

Tira a data pra ficar no padrao.

Reply:
ou deixa tudo com data

Reply:
mas se pa por ser um template faz mais sentido sem data?

Reply:
Particularmente acho que data nunca faz sentido.

Reply:
ah, eu particulamente acho que não faz sentido botar a data tbm não, pq é só usar o gitlens para saber quem foi o último a editar cada linha e quando. Acho que vale a pena botar a contato da comp da thunder, para facilitar alguém de fora do ciclo thunder/usp tirar alguma duvida:

```suggestion
/**
 * @brief HAL GPIO class header
 *
 * @author Thunderatz Development Team <comp@thunderatz.org>
 */
```

Historicamente na equipe eu sempre senti que a gente sempre usou uns headers que nao tem mt utilidade, tipo `@file gpio.hpp`.  Por mim, eh soh ler o nome do arquivo que vc estah mexendo tlgd haha

Alem disso, @brief  eh meio generico mt vezes, se for para falar algo mt obvio, acho q nem precisar colocar ele. Eu pessoalmente gosto da como os [arquivos do BehaviorTree.CPP](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/master/include/behaviortree_cpp/controls/reactive_fallback.h) sao documentados.

Reply:
se n me engano o doxygen quebra qnd nao tem o `@file`, mas concordo com o resto. Gosto do author thunderatz